src/cli.rs

```diff
@@ -60,143 +113,135 @@ pub fn cli_init() -> AppConfig {
                 .long("log-to-file")
                 .help("Enables logging of events to a file located in .paperoni/logs with a default log level of debug. Use -v to specify the logging level")
                 .takes_value(false));
-    let arg_matches = app.get_matches();
 
-    let mut urls: Vec<String> = match arg_matches.value_of("file") {
-        Some(file_name) => {
-            if let Ok(mut file) = File::open(file_name) {
-                let mut content = String::new();
-                match file.read_to_string(&mut content) {
-                    Ok(_) => content
-                        .lines()
-                        .filter(|line| !line.is_empty())
-                        .map(|line| line.to_owned())
-                        .collect(),
-                    Err(_) => vec![],
-                }
-            } else {
-                println!("Unable to open file: {}", file_name);
-                vec![]
-            }
-        }
-        None => vec![],
-    };
-
-    if let Some(vals) = arg_matches.values_of("urls") {
-        urls.extend(
-            vals.filter(|val| !val.is_empty())
-                .map(|val| val.to_string()),
-        );
+        Self::try_from(app.get_matches())
     }
 
-    let max_conn = arg_matches
-        .value_of("max_conn")
-        .map(|conn_str| conn_str.parse::<usize>().ok())
-        .flatten()
-        .map(|max| if max > 0 { max } else { 1 })
-        .unwrap_or(8);
-
-    let mut app_config = AppConfig::new(max_conn);
-    app_config.set_urls(urls);
-
-    if let Some(name) = arg_matches.value_of("output_name") {
-        let file_path = Path::new(name);
-        if file_path.is_dir() {
-            eprintln!("{:?} is a directory", name);
-            std::process::exit(1);
-        }
-
-        let file_name = if file_path.extension().is_some() {
-            name.to_owned()
-        } else {
-            name.to_owned() + ".epub"
-        };
-
-        match std::fs::File::create(&file_name) {
-            Ok(_) => (),
-            Err(e) => {
-                eprintln!("Unable to create file {:?}\n{}", file_path, e);
-                std::process::exit(1)
-            }
-        }
-        app_config.merged = Some(file_name);
+    fn init_merge_file(self) -> Result<Self, Error> {
+        self.merged
+            .as_deref()
+            .map(fs::File::create)
+            .transpose()
+            .err()
+            .map(|err| Err(Error::InvalidOutputPath(err.to_string())))
+            .unwrap_or(Ok(self))
     }
 
-    if arg_matches.is_present("verbosity") {
-        if !arg_matches.is_present("log-to-file") {
-            app_config.can_disable_progress_bar = true;
+    fn init_logger(self) -> Result<Self, Error> {
```

Comment:
This function should still remain in the `log` crate but then return `Result<(),Error>`

Reply:
If you don't mind, I will transfer the pure functionality to the crate, leaving this function here.

Reply:
Okay. My mistake for not noticing that the builder pattern being used needs it to be this way.